pkg/controller/fileobserver/observer_linux.go

```diff
@@ -0,0 +1,68 @@
+// +build linux
+package fileobserver
+
+import (
+	"github.com/golang/glog"
+	"github.com/sigma/go-inotify"
+)
+
+type ObserverLinux struct {
+	watcher   *inotify.Watcher
+	reactors  map[string][]reactorFn
+	isRunning bool
+}
+
+func init() {
+	NewObserver = NewLinuxObserver
+}
+
+func NewLinuxObserver() (Observer, error) {
+	w, err := inotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+	return &ObserverLinux{watcher: w}, nil
+}
+
+func (o *ObserverLinux) AddReactor(reactor reactorFn, filenames ...string) Observer {
+	for _, f := range filenames {
+		glog.V(5).Infof("adding file %q", f)
+		o.watcher.AddWatch(f, inotify.IN_CREATE|inotify.IN_DELETE|inotify.IN_MODIFY)
```

Comment:
only call AddWatch if not yet registered.

Reply:
@sttts what if you want multiple reaction to a single file? ;-)